### PR TITLE
fix ID2= debug tag explosion

### PR DESCRIPTION
**Pull request by maxtaco**

### CODE REVIEW

File: go/engine/context.go [hunk 1]
```diff
@@ -71,3 +71,8 @@ func (c *Context) CloneGlobalContextWithLogTags(g *libkb.GlobalContext, k string
 	c.NetContext = netCtx
 	return g.CloneWithNetContextAndNewLogger(netCtx)
 }
+
```
Comment (patrickxb) [inline go/engine/context.go]: line 55 should be context.Background()

**Comment by patrickxb** [comment]
I don't really understand this, but wasn't involved in the original development/review.

Why a shallow copy instead of just a new context?

**Comment by maxtaco** [comment]
I want to have a mega-debug tag for the background identifier loop itself `BG=3se334EeFe` or so.

If we don't copy the Context over, we'll be continuously accumulating tags [here](https://github.com/keybase/client/blob/7b0d8e272a7fcca9c16e194ed606b58621d00951/go/engine/context.go#L70-L71)

**Comment by maxtaco** [comment]
Usually that's OK before we throw these `engine.Context`'s away after using them once.

**Comment by patrickxb** [comment]
Ok, it still seems like you could do something along the lines of

    bctx := context.WithValue(ctx.NetContext, "BG", "3se334EFe")
    for _, uid := range users {
        ctx.NetContext = context.WithValue(bctx, "ID", uid)
        engine.Run(id2eng, ctx)
    }

Then the `net/context` would only be around for a single run of the id2 engine.

If what you have here works, then I'm ok with it.  I think I get confused reading diff with `engine.Context` and `net/context.Context`.

**Comment by maxtaco** [comment]
The identify2 engine adds its own `ID2` tag internally.  I'll switch things around touhg.

**Comment by maxtaco** [comment]
this any better?

**Comment by patrickxb** [comment]
Yes, that is easier to understand (at least for me!)  thanks.

**Comment by maxtaco** [comment]
Cool, thanks for the the help.  This still is already too complicated due to all of the hacks.  Would be much easier to just pass Context's everywhere!

**Comment by patrickxb** [comment]
Yeah, unfortunately engine.Context predates `net/context` too.

**Pull request merged**